palettes.h

```diff
@@ -808,6 +810,7 @@ struct PaletteGroups {
 			case PaletteGroup::Canon: return Canon;
 			case PaletteGroup::NonCanon: return NonCanon;
 			case PaletteGroup::All: return All;
+			default: throw std::domain_error("PaletteGroup " + std::to_string((int) group) + " is not valid.");

```

Comment:
I am not necessarily making a decision on #41 here, I just wanted to do something semi-reasonable to suppress the missing branch error.